src/ucp/core/ucp_request.inl

```diff
@@ -467,20 +464,13 @@ ucp_request_send_state_advance(ucp_request_t *req,
             /* cppcheck-suppress nullPointer */
             req->send.state.dt        = *new_dt_state;
         }
-
-        if (UCS_STATUS_IS_ERR(status)) {
-            /* fast-forward multi-fragment protocol */
-            req->send.state.dt.offset = req->send.length;
-        }
-
-        if (proto != UCP_REQUEST_SEND_PROTO_BCOPY_AM) {
-            ucp_request_send_state_advance_comp(req, status);
-        }
-
-        break;
+        /* Fall through */
     case UCP_REQUEST_SEND_PROTO_RMA:
-        ucp_request_send_state_advance_comp(req, status);
+        if (status == UCS_INPROGRESS) {
+            ++req->send.state.uct_comp.count;
+        }
         break;
+    case UCP_REQUEST_SEND_PROTO_BCOPY_AM:
```

Comment:
maybe better print proto number in `fatal` message? What is the point to have `UCP_REQUEST_SEND_PROTO_BCOPY_AM` here?

Reply:
done